app/models/message.rb

```diff
@@ -126,41 +68,30 @@ def id_and_code
   end
 
   def reply_url
-    if group
-      "#{Setting.get(:url, :site)}messages/#{id}"
-    else
-      reply_subject = subject
-      reply_subject = "RE: #{subject}" unless reply_subject =~ /^re:/i
-      "#{Setting.get(:url, :site)}messages/new?to_person_id=#{person.id}?subject=#{URI.escape(reply_subject)}"
-    end
+    return "#{Setting.get(:url, :site)}messages/#{id}" if group
+    reply_subject = subject =~ /^re:/i ? subject : "RE: #{subject}"
+    "#{Setting.get(:url, :site)}messages/new?to_person_id=#{person.id}?subject=#{URI.escape(reply_subject)}"
   end
 
   def reply_instructions(to_person)
     msg = []
-    if to || group
-      msg << I18n.t('messages.email.reply_to_sender', person: person.try(:name))
-    end
-    if group
-      if group.can_post?(to_person)
-        if group.address.present?
-          msg << I18n.t('messages.email.reply_to_group', group: group.try(:name), address: group.try(:full_address))
-        else
-          msg << I18n.t('messages.email.reply_link', url: reply_url)
-        end
-      end
-      msg << I18n.t('messages.email.group_link', url: "#{Setting.get(:url, :site)}groups/#{group.id}")
-    end
-    msg.join("\n") + "\n"
+    msg << I18n.t('messages.email.reply_to_sender', person: person.try(:name)) if to || group
+    (msg << case
+            when group.address.present?
+              I18n.t('messages.email.reply_to_group',
+                     group:   group.try(:name),
+                     address: group.try(:full_address))
+            else
+              I18n.t('messages.email.reply_link',
+                     url: reply_url)
+            end) if group && group.can_post?(to_person)
+    msg << I18n.t('messages.email.group_link', url: "#{Setting.get(:url, :site)}groups/#{group.id}") if group
+    msg.join("\n") << "\n"
```

Comment:
Readability of this method went way down IMO. In many simple cases, I'm ok with using a postfix conditional, but in this case I think it makes the method very hard to understand. :cry: